source/opt/inline_pass.cpp

```diff
@@ -620,13 +620,19 @@ bool InlinePass::GenInlineCode(
     AddLoad(calleeTypeId, resId, returnVarId, &new_blk_ptr,
             call_inst_itr->dbg_line_inst(), call_inst_itr->GetDebugScope());
   } else {
-    // Even though it is very unlikely, it is possible that the result id of
-    // the void-function call is used, so we need to generate an instruction
-    // with that result id.
-    std::unique_ptr<Instruction> undef_inst(
-        new Instruction(context(), SpvOpUndef, call_inst_itr->type_id(),
-                        call_inst_itr->result_id(), {}));
-    context()->AddGlobalValue(std::move(undef_inst));
+    // If the function does not return a value, there could be OpName
```

Comment:
This misses decorations, but also I think this is unnecessary. When the function call is removed it should remove all associated names and decorations.

Reply:
Hmm - it doesn't.  I changed `InlineCallWithStructArgContainingSampledImage` in inline_opaque_test to give a name to the result id of a call, and without the code I added here the name doesn't get removed and the module becomes invalid.

It wasn't that clear to me where in the inliner the function call instruction actually gets killed.  `GenInlineCode` has a `call_inst_itr` parameter which is the call instruction, but I don't see where this instruction gets explicitly eliminated.

Reply:
Exhaustive inlining seems to call KillInst. We should probably unify how the various inliners deal with this.

Reply:
OK, I have unified this by moving the code that kills names and decorations into GenInlineCode, which is called by both inliners.  That indeed avoids the need for the extra code above.